hack/deploy-metrics-adapter.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+
+REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+source "${REPO_ROOT}"/hack/util.sh
+function usage() {
+  echo "This script will deploy karmada-metrics-adapter on host cluster"
+  echo "Usage: hack/deploy-metrics-adapter.sh  <HOST_CLUSTER_KUBECONFIG> <HOST_CLUSTER_NAME>>"
```

Comment:
I think we need to specify the karmada-apiserver kubeconfig and context name

Reply:
The karmada-apiserver context is causing too much confusion. I think the kubeconfig would be the same as the host-cluster-kubeconfig and the context karmada-apiserver is being used as sort of control parameter for permissions and security purpose.

Reply:
The point is, karmada-apiserver is just an extension of kube-apiserver and in that case, even if we had a single context only for the HOST_CLUSTER, wouldn't it still work in the same way?

Reply:
The architecture looks like K8s on K8s:
![image](https://github.com/karmada-io/karmada/assets/19745536/f31d51f3-8652-4da2-86d9-c6967d50b3c5)

karmada-apiserver context is used to connect to the kube-apiserver in karmada-system.
karmada-host is used to connect to the kube-apiserver in the host cluster.

Reply:
Ah, this makes more sense. It is namespaced in the same cluster for security and flexibility purposes.